tools/generator/generator.py

```diff
@@ -23,7 +26,9 @@ def template(self, name, out_name=None, **params):
             out_name = name
         tpl = self.env.get_template(name + '.jinja2')
         out = tpl.stream(game=self.game, **params)
-        out.dump((self.out_dir / out_name).open('w'))
+        out_path = self.out_dir / out_name
+        out_path.parent.mkdir(parents=True, exist_ok=True)
+        out.dump(out_path.open('w'))
```

Comment:
(I just realized there's a fd leaking here, although it's not because of this PR)